qiita_pet/handlers/api_proxy/artifact.py

```diff
@@ -267,41 +269,55 @@ def artifact_post_req(user_id, filepaths, artifact_type, name,
     if access_error:
         return access_error
 
-    uploads_path = get_mountpoint('uploads')[0][1]
-    path_builder = partial(join, uploads_path, str(study_id))
-    cleaned_filepaths = []
-    for ftype, file_list in viewitems(filepaths):
-        # JavaScript sends us this list as a comma-separated list
-        for fp in file_list.split(','):
-            # JavaScript will send this value as an empty string if the
-            # list of files was empty. In such case, the split will generate
-            # a single element containing the empty string. Check for that case
-            # here and, if fp is not the empty string, proceed to check if
-            # the file exists
-            if fp:
-                # Check if filepath being passed exists for study
-                full_fp = path_builder(fp)
-                exists = check_fp(study_id, full_fp)
-                if exists['status'] != 'success':
-                    return {'status': 'error',
-                            'message': 'File does not exist: %s' % fp}
-                cleaned_filepaths.append((full_fp, ftype))
-
-    # This should never happen, but it doesn't hurt to actually have
-    # a explicit check, in case there is something odd with the JS
-    if not cleaned_filepaths:
-        return {'status': 'error',
-                'message': "Can't create artifact, no files provided."}
+    if artifact_id:
+        # if the artifact id has been provided, import the artifact
+        try:
+            artifact = Artifact.copy(Artifact(artifact_id), prep)
+        except Exception as e:
```

Comment:
This pattern seems to be replicated in two different locations, the reality of it is that if anything goes wrong, we want the interface to know about that, so perhaps the thing to do here is to follow the decorator pattern and decorate these methods to catch a general exception and return the appropriate dictionary with the encoded information we need? Not blocking, but I thought I would bring this up now.